annotations/builder/src/main/java/io/sundr/builder/internal/functions/ClazzAs.java

```diff
@@ -254,6 +270,37 @@ public TypeDef apply(RichTypeDef item) {
                   .accept(new AddNoArgWithMethod())
                   .build());
     }
+
+    private Method createDescendantBuilderMethod(Set<Property> allDescendants) {
+      BlockNested<MethodBuilder> builderBuilder = new MethodBuilder().withNewModifiers().withProtected().withStatic()
+          .endModifiers()
+          .withParameters(Types.T)
+          .withArguments(new PropertyBuilder().withName("item").withTypeRef(Types.OBJECT_REF).build())
```

Comment:
You can use `addNewArgument` here to avoid manually creating the property builder.